src/resolvers/Mutation/createDirectChat.ts

```diff
@@ -50,7 +50,7 @@ export const createDirectChat: MutationResolvers["createDirectChat"] = async (
   const usersInDirectChat = [];
 
   // Loops over each item in args.data.userIds list.
-  for await (const userId of args.data!.userIds) {
+  for await (const userId of args.data?.userIds ?? []) {
```

Comment:
Much of these type errors in mutations are because of an imperfect graphql schema. When you're mutating things, you don't expect a nullable object from client, you expect a required object that has mutation data. Otherwise calling a mutation resolver makes no sense.

Reply:
I made some basic modifications in graphql scheme for `input` and `mutations`. Similar errors were in `updateUserPassword` as well. The args.data will strictly required newPassword, confirmPassword and oldPassword. So the args. data was made a non-nullable field. Similar changes are made